BraveWallet/Crypto/BuySwapSwap/SwapCryptoView.swift

```diff
@@ -78,98 +75,78 @@ struct ShortcutAmountGrid: View {
+    static let twoPercent: Option = .init(value: 0.02)
+    static let predefinedOptions: [Option] = [
+      .halfPercent, .onePercent, .twoPercent
+    ]
   }
   
-  var action: (Double) -> Void
-  @State private var selectedIndex: Int = 0
-  @State private var input = ""
-  
   @Environment(\.sizeCategory) private var sizeCategory
   
-  private var backgroundShape: some InsettableShape {
-    RoundedRectangle(cornerRadius: 10, style: .continuous)
-  }
-  
   var body: some View {
     HStack(spacing: 8) {
-      ForEach(Option.allCases, id: \.rawValue) { option in
-        if case .other = option {
-          TextField(option.label, text: $input, onEditingChanged: { changed in
-            if selectedIndex != option.index && changed {
-              selectedIndex = option.index
-            }
-          })
-            .onChange(of: input) { value in
-              if selectedIndex == option.index {
-                action((Double(value) ?? option.value) / 100)
-              }
-            }
-            .keyboardType(.numberPad)
-            .multilineTextAlignment(.center)
+      ForEach(Option.predefinedOptions, id: \.id) { option in
+        Button(action: {
+          customSlippage = nil
+          selectedSlippage = option
+          resignFirstResponder()
+        }) {
+          Text(option.localizedString)
             .lineLimit(1)
             .minimumScaleFactor(0.75)
             .padding(.vertical, 12)
             .frame(maxWidth: .infinity)
-            .foregroundColor(Color(option.index == selectedIndex ? .white : .secondaryBraveLabel))
-            .modifier(RroundedBackgroundModifier(backgroundColor: Color(option.index == selectedIndex ? .braveLighterBlurple : .secondaryBraveGroupedBackground)))
+            .foregroundColor(Color(isPredefinedOptionSelected(option.id) ? .white : .secondaryBraveLabel))
+            .background(RoundedBackground(backgroundColor: Color(isPredefinedOptionSelected(option.id) ? .braveBlurple : .secondaryBraveGroupedBackground)))
             .padding(.top, 8)
-        } else {
-          Button(action: {
-            if selectedIndex != option.index {
-              selectedIndex = option.index
-            }
-            input = ""
-            action(option.value)
-            resignFirstResponder()
-          }) {
-            Text(option.label)
-              .lineLimit(1)
-              .minimumScaleFactor(0.75)
-              .padding(.vertical, 12)
-              .frame(maxWidth: .infinity)
-              .foregroundColor(Color(option.index == selectedIndex ? .white : .secondaryBraveLabel))
-              .modifier(RroundedBackgroundModifier(backgroundColor: Color(option.index == selectedIndex ? .braveLighterBlurple : .secondaryBraveGroupedBackground)))
-              .padding(.top, 8)
-          }
         }
       }
+      TextField("%", text: $input)
+        .onChange(of: input) { value in
+          customSlippage = Int(value)
```

Comment:
Can we cap this at 100 like desktop does? (and make sure you adjust to remove negative numbers)